### DISCUSSION TITLE
How to check client connection.

### DISCUSSION THREAD

**Discussion by huylong** [aws/aws-sdk-php#2664] [Q&A] [answered]
Sorry.

I'm pulling data from a queue.
But I have to make many connections.
I want to test if cloudwatch connection works.
I won't initialize it.

`$redisQueue = new RedisQueue($cloudWatchConfig['cloud_watch_queue']['database'], $cloudWatchConfig['cloud_watch_queue']['config_redis']);

        while (true) {
            try {
                if (!empty($cloudWatchConfig['sdk_params']) && !empty($cloudWatchConfig['group_list']) && !empty($cloudWatchConfig['cloud_watch_queue'])) {
                    $dataQueue = $redisQueue->get();
                    //echo '====================================OPEN DATA QUEUE ===================================================' .PHP_EOL;
                    //print_r($dataQueue);
                    //echo '====================================OPEN DATA QUEUE===================================================' .PHP_EOL;
                    //echo 'Start Queue CloudWatch: ' . date('H:i:s d/m/Y') .PHP_EOL;
                    
                    if (empty($dataQueue['data']['stream_name'])) {
                        $dataQueue['data']['stream_name'] = !empty($dataQueue['data']['function'])? $dataQueue['data']['function'] : self::STREAM_NAME;
                    }

                    if (isset($dataQueue) && !empty($dataQueue['name_logger']) && !empty($dataQueue['data']) && !empty($dataQueue['data']['stream_name'])) {
                        $client = new Aws\CloudWatchLogs\CloudWatchLogsClient($cloudWatchConfig['sdk_params']);
                        $log = new Monolog\Logger($dataQueue['name_logger']);
                        $streamName = $this->getStreamName($groupList, $dataQueue['data']['stream_name']);
                        $handler = new Maxbanton\Cwh\Handler\CloudWatch($client, $groupList[$streamName]['group_name'], $streamName, $groupList[$streamName]['retention_days'], 1000, ['Repitte' => 'tag-value']);
                        $log->pushHandler($handler);
                        $log->debug(json_encode($dataQueue['data'], true));
                    }
                    
                    //remove queue data
                    $redisQueue->remove($dataQueue);
                    //echo '====================================OPEN RESULT QUEUE ===================================================' .PHP_EOL;
                    //print_r($result);
                    //echo '====================================OPEN RESULT QUEUE===================================================' .PHP_EOL;
                    //echo 'End Queue CloudWatch: ' . date('H:i:s d/m/Y') .PHP_EOL;
                }
            } catch (Exception $e) {
            }
        }`

**Comment by yenfryherrerafeliz** [comment, marked as answer]
Hi @huylong, thanks for opening this discussion. The only way that we provide to know if a connection will work, when using a SDK client and from a SDK perspective, is by actually doing an operation with the client, and this means that you need to initialize it. 

Thanks!

[a comment by an automated account is not shown]
